.github/workflows/Ray_CI.yml

```
@@ -0,0 +1,93 @@
+name: Ray.jl - CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    tags: ["*"]
+    paths: ["Ray.jl/**", ".github/workflows/Ray_CI.yml"]
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths: ["Ray.jl/**",".github/workflows/Ray_CI.yml"]
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+jobs:
+  test:
+    # Run on push's or non-draft PRs
+    if: (github.event_name == 'push') || (github.event.pull_request.draft == false) || (github.event_name == 'workflow_dispatch')
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.threads}} threads - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.8'
+          - '1.9'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
+        threads:
+          - 1
+    env:
+      JULIA_NUM_THREADS: ${{ matrix.threads }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Clone Ray repository
+        uses: actions/checkout@v3
+        with:
+          repository: beacon-biosignals/ray
+          path: deps/ray
+      - name: Determine Ray commit
+        id: ray-commit
+        run: |
+           cd deps/ray
+           head_sha="$(git rev-parse HEAD)"
+           echo "head_sha=$head_sha" | tee -a "$GITHUB_OUTPUT"
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v1
+      # use build cache from jll CI
+      # TODO: We shouldn't require separate caches per Julia version but allow cache restores to work
+      # https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+      - uses: actions/cache/restore@v3
+        id: bazel-cache
+        with:
+          key: bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
+          path: ~/.cache/bazel
+          restore-keys: |
+            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
+            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
+            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.
```

The GHA cache action doesn't like to share caches between concurrently running jobs I would suggest doing:
```suggestion
          key: bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
          path: ~/.cache/bazel
          restore-keys: |
            bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
            bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
            bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.
```

---

ah that's an interesting point.  I'd hoped to actually re-use the jll cache; but i guess if we don't need to do _full_ re-builds then it's probably okay.